pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/TopicStats.java

```diff
@@ -121,7 +121,12 @@ public TopicStats add(TopicStats stats) {
             }
         } else {
             for (String subscription : stats.subscriptions.keySet()) {
-                this.subscriptions.get(subscription).add(stats.subscriptions.get(subscription));
+                if (this.subscriptions.get(subscription) != null) {
```

Comment:
I think we should try to use putIfAbsent to improve the logic here to handle concurrent updates.